sycl/unittests/CMakeLists.txt

```diff
@@ -2,11 +2,15 @@ add_custom_target(PiUnitTests)
 set_target_properties(PiUnitTests PROPERTIES FOLDER "Tests")
 
 function(add_llvm_unittest test_dirname)
+  # Regardless of `LLVM_BUILD_TESTS`, always build SYCL unit tests.
```

Comment:
I think it's important to allow developers to control the scope of built tests.

According to my understanding the intent of this patch is to avoid regressions in PI unittests. This can be achieve by configuring builtbot scripts to enable build and execution of PI unittests.
https://github.com/intel/llvm/blob/sycl/buildbot/configure.py

I suggest either adding `SYCL_BUILD_TESTS` cmake option (similar to LLVM_BUILD_TESTS) to control build of the SYCL tests.
I don't have strong preference on default value, but I think we should have an option to disable build of the tests.

All contributions to the SYCL implementation must pass buildbot testing, so enabling build of the SYCL tests via `-DSYCL_BUILD_TESTS=ON` in https://github.com/intel/llvm/blob/sycl/buildbot/configure.py, should prevent regressions in PI unittests build.

I think we also should enable execution of the PI unittests in CI.

Reply:
Good point. I will follow the way other LLVM projects deal with unittests, i.e.:
* Declare an option if to include unittest targets: `SYCL_INCLUDE_TESTS`.
* If `LLVM_BUILD_TESTS` is set, then all unittests explicitly included by the LLVM projects are build.

Reply:
Sounds good to me.
Thanks!

Reply:
@bader I have updated CMake and the buildbot scripts with my interpretation of your suggestions. Feedback highly welcome!

Reply:
The changes I am about to push will build the SYCL unittests on demand when "building" the `check-sycl` target. `LLVM_BUILD_TESTS` is not required anymore. The commit message contains more details.